backend/src/hatchling/builders/config.py

```diff
@@ -787,7 +787,7 @@ def default_packages(self):
         return []
 
     def default_global_exclude(self):
-        patterns = ['*.py[cdo]', *EXCLUDED_DIRECTORIES]
```

Comment:
These directories are already excluded during traversal so this pattern matching was wasteful